docs/DeveloperGuide.md

```diff
@@ -1,38 +1,252 @@
 # Developer Guide
 
+## Table of Contents
+
+- [Developer Guide](#developer-guide)
+    - [Table of Contents](#table-of-contents)
+    - [Acknowledgements](#acknowledgements)
+    - [Design \& Implementation](#design--implementation)
+        - [UI and I/O](#ui-and-io)
+        - [Commands](#commands)
+        - [Storage](#storage)
+        - [Item and ItemMap](#item-and-itemmap)
+        - [DateTime](#datetime)
+        - [Exceptions and Logging](#exceptions-and-logging)
+    - [Product scope](#product-scope)
+        - [Target user profile](#target-user-profile)
+        - [Value proposition](#value-proposition)
+    - [User Stories](#user-stories)
+    - [Non-Functional Requirements](#non-functional-requirements)
+    - [Glossary](#glossary)
+    - [Instructions for Testing](#instructions-for-testing)
+        - [Manual Testing](#manual-testing)
+        - [JUnit Testing](#junit-testing)
+        - [Text UI Testing](#text-ui-testing)
+    - [Future Enhancements](#future-enhancements)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+<!-- @@author yijiano -->
+
+PILL uses the following tools for development:
+1. [JUnit 5](https://junit.org/junit5/) - Used for testing.
+2. [Gradle](https://gradle.org/) - Used for build automation.
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The project is designed using the Model-View-Controller (MVC) architecture, with the following components:
```

Comment:
Since Model - Controller - View is at a higher level comparing to other classes (e.g. Model contains Item and ItemClass) and they are not really a class as well, you may try to create a hierarchy, just to distinguish between the architecture and the actual implementation. For example:
3. Controller:
3.1. Item
3.2. ItemMap